paasta_tools/kubernetes_tools.py

```diff
@@ -944,7 +944,11 @@ def get_autoscaled_instances(self) -> int:
             return None
 
     def set_autoscaled_instances(self, instance_count: int) -> None:
-        raise NotImplementedError()
+        """Set the number of instances in the same way that the autoscaler does."""
+        kube_client = KubeClient()
```

Comment:
not that I'm aware of. We don't actually use the kube_client until we set the replica number. The current design pattern in `kubernetes_tools.py` is to instantiate functions that talk to kube_client outside of the `KubernetesDeploymentConfig` class and to pass into kube_client as a parameter

Reply:
That is too bad. It would be nice not to have to instantiate a kube client upon every API request.

What is `instance_set_state` doing with `kube_client=settings.kubernetes_client` ?

If this works, can you make the code re-use the existing kube_client that the api already has?

Reply:
@solarkennedy  Makes sense. This would be tricky because `set_autoscaled_instances` is an overridden method from `LongRunningServiceConfig()` which requires the same signature. I'm unsure if we want to decouple this? Also, `get_autoscaled_instances` uses a new instantiated kube client.

Reply:
Ok. Well I recommend functions not instantiate their own clients and instead get one passed in if available.

I think it is fine for each InstanceConfig to have their own implementation of set_autoscaled_instances. To me that is the whole point of having a class abstraction like this.